models/auth/oauth2.go

```diff
@@ -46,6 +48,64 @@ func init() {
 	db.RegisterModel(new(OAuth2Grant))
 }
 
+func Init(ctx context.Context) error {
+	defaultApplications := []*OAuth2Application{
+		{
+			Name:         "git-credential-oauth",
+			ClientID:     "a4792ccc-144e-407e-86c9-5e7d8d9c3269",
+			RedirectURIs: []string{"http://127.0.0.1", "https://127.0.0.1"},
+			Locked:       true,
+		},
+		{
+			Name:         "Git Credential Manager",
+			ClientID:     "e90ee53c-94e2-48ac-9358-a874fb9e0662",
+			RedirectURIs: []string{"http://127.0.0.1", "https://127.0.0.1"},
+			Locked:       true,
+		},
+	}
+
+	registeredOAuth2Apps, _ := GetOAuth2ApplicationsByUserID(ctx, 0)
```

Comment:
Since `client_id` is unique, searching by the `client_id` is the best resolution here. You can use `In` because the default applications should not too much. Gitea supports Global Applications, although there should not be too many global applications.

Reply:
The idea behind that line is that we only retrieve the list once, as these OAuth2 apps are being registered globally. I can change it to a lookup by client_id, but that will increase the number of DB calls.....

Reply:
If that's not a problem, I'd also prefer a client_id lookup

Reply:
You can use client_id IN (client_id1, client_id2) which is only 1 query.

Reply:
Done